### PR TITLE
Spring security

### PR DESCRIPTION
**Pull request by margarita-ignatieva**

### CODE REVIEW

File: src/main/java/com/cinema/shop/validators/EmailValidator.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import javax.validation.ConstraintValidatorContext;
+
+public class EmailValidator implements ConstraintValidator<ValidEmail, String> {
+    private static final String EMAIL_PATTERN =
+            "[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)"
+                    + "*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])";
+
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
+        return email.matches(EMAIL_PATTERN) && email != null;
```
Comment (AnaLanda) [inline src/main/java/com/cinema/shop/validators/EmailValidator.java]:
```suggestion
        return email != null && email.matches(EMAIL_PATTERN);
```

File: src/main/java/com/cinema/shop/validators/ValidPassword.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+@Documented
+@Constraint(validatedBy = PasswordValidator.class)
+@Target({ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface ValidPassword {
+    String message() default "Password repeated is not equal to the password";
```
Comment (AnaLanda) [inline src/main/java/com/cinema/shop/validators/ValidPassword.java]:
```suggestion
    String message() default "Repeated password is not identical to the original password";
```

File: src/main/java/com/cinema/shop/validators/ValidEmail.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+@Documented
+@Constraint(validatedBy = EmailValidator.class)
+@Target({ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface ValidEmail {
+    String message() default "Email is not of a valid representation";
```
Comment (AnaLanda) [inline src/main/java/com/cinema/shop/validators/ValidEmail.java]:
```suggestion
    String message() default "Email is not of a valid format";
```

**Review by AnaLanda — COMMENTED** [review]
Looks good! 😃

File: src/main/java/com/cinema/shop/validators/ValidEmail.java (hunk 3 above)
Comment (dmytro-hryhoruk) [reply src/main/java/com/cinema/shop/validators/ValidEmail.java]: agree

File: src/main/java/com/cinema/shop/config/WebAppInitializer.java [hunk 4]
```diff
@@ -6,7 +6,8 @@
 
     @Override
     protected Class<?>[] getRootConfigClasses() {
-        return new Class[]{AppConfig.class};
+        return new Class[]{
+                AppConfig.class, SecurityConfig.class};
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/shop/config/WebAppInitializer.java]:
```suggestion
        return new Class[]{AppConfig.class, SecurityConfig.class};
```

File: src/main/java/com/cinema/shop/validators/PasswordValidator.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+import com.cinema.shop.model.dto.UserRequestDto;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class PasswordValidator implements ConstraintValidator<ValidPassword, UserRequestDto> {
+
+    @Override
+    public boolean isValid(UserRequestDto userRequestDto,
+                           ConstraintValidatorContext constraintValidatorContext) {
+        return userRequestDto.getPassword().equals(userRequestDto.getRepeatPassword());
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/shop/validators/PasswordValidator.java]: does it work if  userRequestDto.getPassword() == null?

File: src/main/java/com/cinema/shop/validators/EmailValidator.java (hunk 1 above)
Comment (dmytro-hryhoruk) [reply src/main/java/com/cinema/shop/validators/EmailValidator.java]: agree

File: src/main/java/com/cinema/shop/validators/PasswordValidator.java (hunk 5 above)
Comment (margarita-ignatieva) [reply src/main/java/com/cinema/shop/validators/PasswordValidator.java]: I thought if there is a @NotNull annotation above the password than repeat password can't be null

Comment (dmytro-hryhoruk) [reply src/main/java/com/cinema/shop/validators/PasswordValidator.java]: this validation may execute first

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**